client.go

```diff
@@ -480,11 +484,16 @@ func (c *Client) Dial(ctx context.Context) error {
 
 	c.sechan, err = uasc.NewSecureChannel(c.endpointURL, c.conn, c.cfg.sechan, c.sechanErr)
 	if err != nil {
-		_ = c.conn.Close()
+		c.conn.Close()
 		return err
 	}
 
-	return c.sechan.Open(ctx)
+	if err := c.sechan.Open(ctx); err != nil {
+		c.conn.Close()
+		return err
+	}
+
+	return c.UpdateNamespaces()
```

Comment:
Is there ever a case where someone is dialing a server, but does _not_ have permission to read `Server_NamespaceArray`? Or maybe is connecting to a broken server? Maybe this should be gated by an option?

Reply:
This is one of those cases where I will suggest to wait for somebody to actually complain about it or find a broken server somewhere.

Adding an option will add complexity in the interface and we are covering ourselves from a case that it is quite unlikely to happen in my opinion.

Reply:
Oh, missed this comment. Then I'll remove the option again. I'm in favor of keeping it simple.

Reply:
Kept this in a separate commit so reverting it will implement the option :)